tools/ci-operator-prowgen/main_test.go

```diff
@@ -0,0 +1,173 @@
+		podSpec.Containers[0].Env[0].ValueFrom.ConfigMapKeyRef.Key,
+		"branch.json",
+	)
+}
+
+func TestGeneratePodSpecWithAdditionalArgs(t *testing.T) {
+	podSpec := generatePodSpec("organization", "repo", "branch", "target", "--promote", "something")
+	checkString(t,
+		"podSpec.Containers[0].Args",
+		strings.Join(podSpec.Containers[0].Args, " "),
+		"--artifact-dir=$(ARTIFACTS) --target=target --promote something",
+	)
+}
+
+func TestGeneratePresubmitForTest(t *testing.T) {
+	presubmit := generatePresubmitForTest(testDescription{"name", "target"}, "organization", "repo", "branch")
+	checkString(t, "presubmit.Agent", presubmit.Agent, "kubernetes")
+	checkCondition(t, "presubmit.AlwaysRun == true", presubmit.AlwaysRun)
+	checkString(t, "presubmit.Brancher.Branches[0]", presubmit.Brancher.Branches[0], "branch")
+	checkString(t, "presubmit.Context", presubmit.Context, "ci/prow/name")
+	checkString(t, "presubmit.Name", presubmit.Name, "pull-ci-organization-repo-name")
+	checkString(t, "presubmit.RerunCommand", presubmit.RerunCommand, "/test name")
+	checkCondition(t, "presubmit.Spec != nil", presubmit.Spec != nil)
+	checkString(t, "presubmit.Trigger", presubmit.Trigger, "((?m)^/test( all| name),?(\\\\s+|$))")
+	checkCondition(t,
+		"presubmit.UtilityConfig.DecorationConfig.SkipCloning == true",
+		presubmit.UtilityConfig.DecorationConfig.SkipCloning,
+	)
+	checkCondition(t, "presubmit.UtilityConfig.Decorate == true", presubmit.UtilityConfig.Decorate)
+}
+
+func TestGeneratePostsubmitForTest(t *testing.T) {
+	postsubmit := generatePostsubmitForTest(
+		testDescription{"name", "target"},
+		"organization",
+		"repo",
+		"branch",
+		"--promote",
+		"something",
+	)
+	checkString(t, "postsubmit.Agent", postsubmit.Agent, "kubernetes")
+	checkString(t, "postsubmit.Name", postsubmit.Name, "branch-ci-organization-repo-name")
+	checkCondition(t, "postsubmit.Spec != nil", postsubmit.Spec != nil)
+	checkString(t,
+		"postsubmit.Spec.Containers[0].Args",
+		strings.Join(postsubmit.Spec.Containers[0].Args, " "),
+		"--artifact-dir=$(ARTIFACTS) --target=target --promote something",
+	)
+
+	checkCondition(t,
+		"postsubmit.UtilityConfig.DecorationConfig.SkipCloning == true",
+		postsubmit.UtilityConfig.DecorationConfig.SkipCloning,
+	)
+	checkCondition(t, "postsubmit.UtilityConfig.Decorate == true", postsubmit.UtilityConfig.Decorate)
+}
+
+func TestGenerateJobs(t *testing.T) {
+	mockConfig := ciop.ReleaseBuildConfiguration{
+		Tests: []ciop.TestStepConfiguration{
+			ciop.TestStepConfiguration{As: "derTest"},
+			ciop.TestStepConfiguration{As: "leTest"},
+		}}
+	jobConfig := generateJobs(&mockConfig, "organization", "repo", "branch")
+	presubmits := (*jobConfig).Presubmits
+	postsubmits := (*jobConfig).Postsubmits
+
+	checkCondition(t, "len(presubmits) == 1", len(presubmits) == 1)
+	checkCondition(t, "len(postsubmits) == 1", len(postsubmits) == 1)
+	checkCondition(t,
+		"len(presubmits[\"organization/repo\"]) == 2",
```

Comment:
Will be not important if you use the `quality.Semantic.DeepEqual` as mentioned above, but FWIW in Go you can use back-tick quotes to have quote literals without escaping:

    `len(presubmits["organization/repo"]) == 2`,

Reply:
OK, thanks